joomla/joomla-cms
# Allow TinyMCE to respect the template's options

Pull Request for improving the UX and making editing a bit easier.

### Summary of Changes
TinyMCE out of the box comes with some very basic styling. Eg the `Bold` button is just a `<strong>` around the selected text. But the editor gives the ability for further customisations e.g.: attach a class or inline styles etc...

So this PR brings that functionality to Joomla.
It covers both styles and formats (styles are the ones that correspond to the buttons e.g. bold, align left, etc and formats are the ones that will be in the formats dropdown)

### Testing Instructions
Grab the files from https://gist.github.com/dgt41/be9e16814371f43ce99e8dc74ef54e99
and place them in `administrator/templates/isis/css` directory
Edit tinyMCE's configuration and type the relative paths:
![screen shot 2017-03-10 at 13 09 07](https://cloud.githubusercontent.com/assets/3889375/23793230/df57963e-0592-11e7-8c95-d2fbbcd4eab8.png)
Edit an article and observe the code when selecting a format or applying a style (e.g. bold, align right)
![screen shot 2017-03-10 at 12 42 49](https://cloud.githubusercontent.com/assets/3889375/23793277/18a5c94c-0593-11e7-8379-be4f6a4bea3e.png)

### Expected result
#### Styles (standard buttons):
![screen shot 2017-03-10 at 13 14 35](https://cloud.githubusercontent.com/assets/3889375/23793383/98cff804-0593-11e7-90ae-064795a8cf82.png)

#### Formats (Formats dropdown
![screen shot 2017-03-10 at 13 14 03](https://cloud.githubusercontent.com/assets/3889375/23793410/b6b35a0a-0593-11e7-98f3-3f35e42157e3.png)
):

### Documentation Changes Required
Yup!

Inspired by @coolcat-creations post on @brianteeman 's site...
